subsys/bluetooth/mesh/Kconfig

```diff
@@ -360,6 +361,17 @@ config BT_MESH_OOB_AUTH_REQUIRED
 	bool "OOB authentication mandates to use HMAC SHA256"
 	depends on BT_MESH_ECDH_P256_HMAC_SHA256_AES_CCM
 
+config BT_MESH_REDUCED_MAX_OOB_SIZE
+	bool "Reduced maximum OOB authentication size support"
+	depends on BT_MESH_PROV
+	help
+	  The initial maximum OOB authentication size was 8 bytes. The API was designed
+	  accordingly. Later, the maximum OOB authentication size was increased to 32 bytes.
+	  Related API for OOB output\input was changed and became incompatible with the previous one.
+	  Enable this option to enable the previous API and support the initial
+	  maximum OOB authentication size. Option is intended for backward compatibility.
+	  Not recommended for new designs.
```

Comment:
What I don't like here is the usage of words `initial` and `later`. When this `initial` and when this `later` happened in time this paragraph doesn't explain, which doesn't make the picture clear for me as a user. Neither it explains the consequences of using old API other new one.

I think we can mention here that prior the Mesh Specification v1.1 inclusively, the maximum OOB size was 8 bytes.

What about this text:
```
By default, the maximum OOB authentication size is 32 bytes.

Enable this option to use the legacy API which is only compatible with
specification v1.1 and earlier. This limits OOB authentication to 8
bytes maximum.

This option is intended for backward compatibility.
Not recommended for new designs due to reduced security.
```